### DISCUSSION TITLE
Using IntuneCD in an Azure Automation account Runbook

### DISCUSSION THREAD

**Discussion by gmenziesint** [almenscorner/IntuneCD#144] [General]
Has anyone managed to use IntuneCD in a python runbook in an Azure Automation account?

I'm having difficulties with this, my python knowledge is limited (Probably the problem).

If anyone has examples please let me know!

**Comment by almenscorner** [comment]
I haven't tested this myself but how and where would you push the exported files?

As IntuneCD is meant to be run from command line the easiest option would probably be to create a script that uses that to run IntuneCD instead of trying to import as a module in a script and running.

**Reply by almenscorner** [reply]
For example, IntuneCD Monitor is using subprocess to run the backup inside the Docker container: https://github.com/almenscorner/intunecd-monitor/blob/9525fbaae812a4f0d40f33f8e748a57e3fb17765/app/run_intunecd.py#L354-L366

**Reply by gmenziesint** [reply]
Hey Tobias,

The plan so far is to push it to an Azure storage account blob, the files would be saved in a Temp directory until they are moved over. I think the end-goal would be to push it to an Azure DevOps Repo, today I'm more familiar with storage accounts and using PowerShell runbooks so this is where I've started but your DevOps Pipeline would be the end-goal once we're happy with everything.

I think you've answered my question with providing subprocess :) I wasn't familiar with it and I believe it would let me just run a terminal command within Python like below

```
import subprocess
subprocess.run("whoami", shell=True)
```

Thanks for your help.
